### PR TITLE
build: Replacing wildcards with paths

### LINKED ISSUE #67: Finding files using wildcards can lead to unexpected behaviour
It already came up during review of #60: we should stop using wildcards in our Makefiles. Both, for nxdk toolchain files, and in our sample code.

### PR DESCRIPTION
**Pull request by Teufelchen1**
Hi,
closes #67

I replaced all wildcards in all Makefiles I could find. There are still wildcards in these sub-repositories:

- `lib/zlib`
- `lib/pdcib`
- `lib/sdl`
- `lib/net/lwip`

### CODE REVIEW

File: tools/cxbe/Makefile [hunk 1]
```diff
@@ -1,5 +1,16 @@
-INCLUDES := $(wildcard *.h)
-SRCS := $(wildcard *.cpp)
+INCLUDES := AlignPosfix1.h \
+	AlignPrefix1.h \
```
Comment (JayFoxRox) [inline tools/cxbe/Makefile:2]: nit: Weird indentation. Either start it with an empty `\` in the first line, then align all with a tab; or align with spaces to align to the first line.

File: tools/fp20compiler/Makefile [hunk 2]
```diff
@@ -1,7 +1,15 @@
 MAIN = fp20compiler
 CXXFLAGS= -DYY_NO_UNPUT -Wall -g
 
-INCLUDES = $(wildcard *.h) \
+INCLUDES = nvparse_errors.h \
```
Comment (JayFoxRox) [inline tools/fp20compiler/Makefile:4]: nit: Weird indentation. Start this with an empty first line which just has `\`. Otherwise it's easy to miss "nvparse_errors.h"

(Other instances, too)

**Comment by JayFoxRox** [comment]
nit: You should consider splitting this into separate commits in case we have to revert a change. Personally I wouldn't do it, but it would have been nice maybe (not too detailed, maybe just `pbkit:` `tools` and `samples:`).

nit: I think the usual commit style would be a command-style / future tense (what this will do; not what it is doing): "build: Replace wildcards by explicit path".


Other than that it looked fine during a first review pass. I'll probably build this myself from scratch to make sure nothing was forgotten.

File: samples/mesh/Makefile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 XBE_TITLE = mesh
 GEN_XISO = $(XBE_TITLE).iso
-SRCS = $(wildcard $(CURDIR)/*.c)
+SRCS += $(CURDIR)/math3d.c $(CURDIR)/main.c
```
Comment (thrimbor) [inline samples/mesh/Makefile:3]: Why the change to `+=`?

File: tools/fp20compiler/Makefile [hunk 4]
```diff
@@ -1,12 +1,22 @@
...
+INCLUDES = \
+	nvparse_errors.h \
+	nvparse_externs.h \
+	ps1.0_program.h \
+	rc1.0_combiners.h \
+	rc1.0_final.h \
+	rc1.0_general.h \
+	rc1.0_register.h \
+	ts1.0_inst.h \
+	ts1.0_inst_list.h\
```
Comment (thrimbor) [inline tools/fp20compiler/Makefile:13]: nit: No space before `\`

File: samples/sdl/Makefile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 XBE_TITLE = sdl
 GEN_XISO = $(XBE_TITLE).iso
-SRCS += $(wildcard $(CURDIR)/*.c)
+SRCS += $(CURDIR)/main.c
```
Comment (JayFoxRox) [inline samples/sdl/Makefile:3]: This one used to use `+=`, but I think this should just be `=` to be more consistent with other samples.

The httpd sample is also affected by this.

File: samples/mesh/Makefile (hunk 3 above)
Comment (JayFoxRox) [reply samples/mesh/Makefile:3]: You pushed since this review, but this one isn't fixed yet? @Teufelchen1

**Comment by thrimbor** [comment]
Looks good now, merging.
